# NRC Emotional Lexicon

Αυτό είναι το [NRC Emotional Lexicon](http://saifmohammad.com/WebPages/NRC-Emotion-Lexicon.htm): "The NRC Emotion Lexicon is a list of English words and their associations with eight basic emotions (anger, fear, anticipation, trust, surprise, sadness, joy, and disgust) and two sentiments (negative and positive). The annotations were manually done by crowdsourcing."



In [5]:
import pandas as pd

In [6]:
filepath = "NRC-emotion-lexicon-wordlevel-alphabetized-v0.92.txt"
emolex_df = pd.read_csv(filepath,  names=["word", "emotion", "association"], skiprows=45, sep='\t')
emolex_df.head()

word       emotion  association
0  aback         anger            0
1  aback  anticipation            0
2  aback       disgust            0
3  aback          fear            0
4  aback           joy            0

Είναι αρκετά εύκολο. Μια στήλη με τη λέξη, μια για το συναίσθημα και μία για την συσχέτιση.  
Βλέπετε το "aback aback aback aback" επειδή υπάρχει μια γραμμή για κάθε ζευγάρι λέξης-συναισθήματος. 

## Ποια συναισθήματα υπάρχουν?


In [7]:
emolex_df.emotion.unique()

array(['anger', 'anticipation', 'disgust', 'fear', 'joy', 'negative',
       'positive', 'sadness', 'surprise', 'trust'], dtype=object)

In [8]:
emolex_df.emotion.value_counts()

joy             14182
trust           14182
anticipation    14182
fear            14182
negative        14182
anger           14182
surprise        14182
sadness         14182
positive        14182
disgust         14182
Name: emotion, dtype: int64

## Πόσες λέξεις έχει κάθε συναίσθημα?

Κάθε συναίσθημα δεν έχει 14182 λέξεις που σχετίζονται με αυτό, αλλά το  `1` σημαίνει "σχετίζεται" και `0` σημαίνει "δεν σχετίζεται"


In [9]:
emolex_df[emolex_df.association == 1].emotion.value_counts()

negative        3324
positive        2312
fear            1476
anger           1247
trust           1231
sadness         1191
disgust         1058
anticipation     839
joy              689
surprise         534
Name: emotion, dtype: int64

## Ποιες λέξεις δείχνουν θυμό?

In [10]:
emolex_df[(emolex_df.association == 1) & (emolex_df.emotion == 'anger')].word

30          abandoned
40        abandonment
170             abhor
180         abhorrent
270           abolish
300       abomination
630             abuse
1120         accursed
1130       accusation
1150          accused
1160          accuser
1170         accusing
1470       actionable
1650            adder
2390        adversary
2400          adverse
2410        adversity
2500         advocacy
2840          affront
2920        aftermath
3030       aggravated
3040      aggravating
3050      aggravation
3080       aggression
3090       aggressive
3100        aggressor
3140         agitated
3150        agitation
3190            agony
3570       alcoholism
             ...     
138470        warlike
138530           warp
138600        warrior
138680         wasted
138690       wasteful
139330          wench
139550           whip
139950        willful
140020          wimpy
140030          wince
140220       wireless
140290          witch
140300     witchcraft
140610            wop
140640    

## Μπορούμε να αλλάξουμε τον τρόπο που εμφανίζονται οι στήλες και οι γραμμές


In [11]:
emolex_words = emolex_df.pivot(index='word', columns='emotion', values='association').reset_index()
emolex_words.head()

emotion       word  anger  anticipation  disgust  fear  joy  negative  \
0              NaN      0             0        0     0    0         0   
1            aback      0             0        0     0    0         0   
2           abacus      0             0        0     0    0         0   
3          abandon      0             0        0     1    0         1   
4        abandoned      1             0        0     1    0         1   

emotion  positive  sadness  surprise  trust  
0               0        0         0      0  
1               0        0         0      0  
2               0        0         0      1  
3               0        1         0      0  
4               0        1         0      0

Επίσης, μπορούμε να ψάξουμε συγκεκριμένες λέξεις...

In [12]:

emolex_words[emolex_words.word == 'charitable']

emotion        word  anger  anticipation  disgust  fear  joy  negative  \
2002     charitable      0             1        0     0    1         0   

emotion  positive  sadness  surprise  trust  
2002            1        0         0      1

...ή συγκεκριμένα συναισθήματα...

In [38]:
emolex_words[emolex_words.anger == 1].head()

emotion         word  anger  anticipation  disgust  fear  joy  negative  \
3          abandoned      1             0        0     1    0         1   
4        abandonment      1             0        0     1    0         1   
17             abhor      1             0        1     1    0         1   
18         abhorrent      1             0        1     1    0         1   
27           abolish      1             0        0     0    0         1   

emotion  positive  sadness  surprise  trust  
3               0        1         0      0  
4               0        1         1      0  
17              0        0         0      0  
18              0        0         0      0  
27              0        0         0      0

...ή πολλά συναισθήματα μαζί!

In [39]:
emolex_words[(emolex_words.joy == 1) & (emolex_words.negative == 1)].head()

emotion        word  anger  anticipation  disgust  fear  joy  negative  \
61        abundance      0             1        1     0    1         1   
1018           balm      0             1        0     0    1         1   
1382     boisterous      1             1        0     0    1         1   
1916      celebrity      1             1        1     0    1         1   
2004        charmed      0             0        0     0    1         1   

emotion  positive  sadness  surprise  trust  
61              1        0         0      1  
1018            1        0         0      0  
1382            1        0         0      0  
1916            1        0         1      1  
2004            1        0         0      0

Το χρήσιμο κομμάτι είναι να χρησιμοποιήσουμε λέξεις που αντιστοιχούν σε **ένα συναίσθημα**

In [13]:
# Angry words
emolex_words[emolex_words.anger == 1].word

4          abandoned
5        abandonment
18             abhor
19         abhorrent
28           abolish
31       abomination
64             abuse
113         accursed
114       accusation
116          accused
117          accuser
118         accusing
148       actionable
166            adder
240        adversary
241          adverse
242        adversity
251         advocacy
285          affront
293        aftermath
304       aggravated
305      aggravating
306      aggravation
309       aggression
310       aggressive
311        aggressor
315         agitated
316        agitation
320            agony
358       alcoholism
            ...     
13847        warlike
13853           warp
13860        warrior
13868         wasted
13869       wasteful
13933          wench
13955           whip
13995        willful
14002          wimpy
14003          wince
14022       wireless
14029          witch
14030     witchcraft
14061            wop
14064          words
14087      worthless
14090        